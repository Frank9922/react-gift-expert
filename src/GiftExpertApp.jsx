import { useState } from "react"
import { AddCategory, GifGrid , NotCategories} from "./components";


export const GiftExpertApp = ( ) => {

    const [categories, setCategories] = useState( ['One Punch'] );
    const mostrarGifGrid = categories && categories.length !== 0;


    const onAddCategory = (NewCategory) => {
        
        if(categories.includes(NewCategory)) return

        setCategories([NewCategory, ...categories])
    }

    const onRemoveCategory = (Category) => {

        setCategories(categories.filter((c) => c !== Category))
        console.log(categories)
    }


    return (
        <>
            <h1>Gift Expert App</h1>

            <AddCategory 
                onNewCategory={(value) => onAddCategory(value)}
            />

                    {   
                        mostrarGifGrid ? (
                        
                        <div>
                            {categories.map( ( category ) => (

                                <GifGrid
                                    key={category}
                                    category={category}
                                    RemoveCategory={(value) => onRemoveCategory(value)}
                                />
                            ))}
                        </div>
                    ) :
                        <NotCategories />
                    }
            </>

    )

}