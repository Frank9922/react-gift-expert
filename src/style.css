* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(234, 234, 234);
  }
  
  body {
    padding: 60px;
  }
  
  input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 3rem;
    margin-bottom: 5px;
  }
  
  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
  .card-title {
    display: flex;
    justify-content: space-between;
    align-items: baseline;

  }

  .btn-eliminar {
    font-size: large;
    padding: 0.5rem 1.2rem;
    border-radius: 5px;
    background-color: #ffff;
    transition: all 0.1s ease-in-out;
  }
  .btn-eliminar:hover {
    cursor: pointer;
    background-color: #e0e0e0;

  }
  .not-categories {
    display: flex;
    justify-content: center;
    padding: 10% 0;
  }
  .card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }
  
  .card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }