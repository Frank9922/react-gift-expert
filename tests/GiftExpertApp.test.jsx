import { render, screen } from "@testing-library/react"
import { GiftExpertApp } from "../src/GiftExpertApp"

describe('Pruebas en el componente <GiftExpertApp />', () => {


    test('Match componente vacio', () => {

        const { container } = render(<GiftExpertApp />);

        screen.debug();
        expect(container).toMatchSnapshot();
    })
})