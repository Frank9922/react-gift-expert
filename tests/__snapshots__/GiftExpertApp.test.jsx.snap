// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GiftExpertApp /> Match componente vacio 1`] = `
<div>
  <h1>
    Gift Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
